src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1804,6 +1804,7 @@ public ImmutableArray<DocumentId> GetDocumentIdsWithFilePath(string? filePath)
             var map = projectStates.Values.Select(state => new KeyValuePair<ProjectId, ImmutableHashSet<ProjectId>>(
                     state.Id,
                     state.ProjectReferences.Where(pr => projectStates.ContainsKey(pr.ProjectId)).Select(pr => pr.ProjectId).ToImmutableHashSet()))
+                    .Where(pair => !pair.Value.IsEmpty)
```

Comment:
Should this filtering be done on the internal constructor? Otherwise this is more or less leaking an implementation detail outside the class.

Reply:
➡️ Included in #41288